app/stores/ada/HWVerifyAddressStore.js

```diff
@@ -59,33 +66,32 @@ export default class AddressesStore extends Store {
     this._setError(null);
     this._setActionProcessing(true);
 
-    try {
-      // TODO: don't use hardcoded strings and maybe find a better way to do this?
-      // $FlowFixMe
-      if (params.wallet.hardwareInfo.vendor === 'ledger.com') {
-        await this.ledgerVerifyAddress(path);
-      } else if (params.wallet.hardwareInfo.vendor === 'trezor.io') {
-        // not implemented yet
-        await this.trezorVerifyAddress(path);
-      } else {
-        throw new Error('AddressStore::_verifyAddress called with unrecognized hardware wallet');
-      }
-    } catch (error) {
-      Logger.error('AddressStore::_verifyAddress::error: ' + stringifyError(error));
-      this._setError(convertToLocalizableError(error));
-    } finally {
-      this._setActionProcessing(false);
+    if (params.wallet.isLedgerNanoSWallet) {
+      await this.ledgerVerifyAddress(path);
+    } else if (params.wallet.isTrezorTWallet) {
+      await this.trezorVerifyAddress(path, address);
+    } else {
+      throw new Error('AddressStore::_verifyAddress called with unrecognized hardware wallet');
```

Comment:
@SebastienGllmt I just realized after merging this PR that this throw is not catch anywhere and it shouldn’t because it’s an action connected directly to a listener. It should probably be changed to a Logger